simphony/libraries/sipann.py

```diff
@@ -2,40 +2,690 @@
 # Licensed under the terms of the MIT License
 # (see simphony/__init__.py for details)
 
-"""
-simphony.libraries.sipann
-==============
+from simphony import Model
+from simphony.tools import freq2wl
+from SiPANN import scee, comp
 
 
-"""
+class SipannWrapper(Model):
+    """Allows wrapping models from SCEE for use in simphony.
+    This class should be extended, with each extending class
+    wrapping one model.
 
-from simphony import Model
+    Each extending class should convert parameters passed in
+    from meters (which simphony uses) to nanometers (which
+    SiPANN uses).
+    
+    Note that the wrapped SCEE models cannot have varying
+    geometries; such a device can't be cascaded properly.
+
+    Parameters
+    -----------
+    `model`
+    Model from `SiPANN.scee` or `SiPANN.comp` modules
+    
+    `pin_count : int`
+    Number of pins for this model
+
+    `sigmas : dict`
+    Dictionary mapping parameters to sigma values for
+    Monte-Carlo simulations, values should be in meters. If
+    Monte-Carlo simulations are not needed, pass in
+    an empty dictionary.
+    """
+
+    freq_range = (
+        182800279268292.0,
+        205337300000000.0,
+    )
+
+    def __init__(self, model, sigmas):
+        super().__init__()
+        
+        self.model = model
+        self.sigmas = sigmas
+
+        # catch varying geometries
+        args = self.model._clean_args(None)
+        if len(args[0]) != 1:
+            raise ValueError(
+                "You have changing geometries, use in simphony doesn't make sense!"
+            )
+        
+        self.params = self.model.__dict__.copy()
+        self.rand_params = dict()
+        self.regenerate_monte_carlo_parameters()
+
+    def s_parameters(self, freq):
+        """Get the s-parameters of the SCEE Model.
+
+        Parameters
+        ----------
+        `freq : np.ndarray`
```

Comment:
Previous code called the parameter `freq` but it actually contains an array of frequencies.
Make sure to update all `freq` parameters to say `freqs` to conform to the recent changes.

Also note that it has the type of `np.array` not `np.ndarray`.
In fact it looks like many of the typings are set to `np.ndarray` when in fact they should be `np.array`. Make sure to look through all of the typings and make sure they're correct.